api_tests/lib/actor.ts

```diff
@@ -1,39 +1,72 @@
-const Toml = require("toml");
-const wallet = require("./wallet.js");
-const fs = require("fs");
-const bitcoin = require("./bitcoin.js");
-const ethutil = require("ethereumjs-util");
-
-// TODO: pass value instead of using global config
-const bitcoin_rpc_client = bitcoin.createClient(
-    global.harness.ledgers_config.bitcoin
-);
-
-class Actor {
-    constructor(name, config) {
-        const node_config = global.harness.config.comit_node[name];
-        if (!node_config) {
-            throw new Error("comit_node." + name + " configuration is needed");
-        }
+import { WalletConfig, Wallet } from "./wallet";
+import * as chai from "chai";
+import {
+    Action,
+    SwapResponse,
+    ComitNodeConfig,
+    MetaComitNodeConfig,
+} from "./comit";
+import * as bitcoin from "./bitcoin";
+import * as toml from "toml";
+import * as fs from "fs";
+
+import chaiHttp = require("chai-http");
+
+chai.use(chaiHttp);
+
+export interface BtsieveForComitNodeConfig {
+    poll_interval_secs: number;
+}
+
+export interface TestConfig {
+    comit_node: { [key: string]: MetaComitNodeConfig };
+}
+
+export class Actor {
+    name: string;
+    host: string;
+    wallet: Wallet;
+    comitNodeConfig: ComitNodeConfig;
+
+    constructor(
+        name: string,
+        testConfig?: TestConfig,
+        root?: string,
+        walletConfig?: WalletConfig
+    ) {
         this.name = name;
-        this.host = node_config.host;
-        this.config = Toml.parse(
-            fs.readFileSync(
-                global.harness.test_root +
-                    "/" +
-                    node_config.config_dir +
-                    "/default.toml",
-                "utf8"
-            )
-        );
-        this.wallet = wallet.create(name, config);
+        if (testConfig) {
+            const metaComitNodeConfig = testConfig.comit_node[name];
```

Comment:
❓ Why **meta**ComitNodeConfig?

Reply:
Because it's not the actual comit node config.
This refers to the `[comit_node.alice]` section in the `config.toml` file that is with the test.

The section look like that:
```
[comit_node.bob]
host = "127.0.0.1"
config_dir = "./regtest/bob/"
```

The `config_dir` contains the path to the directory that contains the actual configuration that comit-rs uses.

Reply:
Thanks for the clarification.